app/src/main/java/com/example/peterchu/watplanner/home/HomeFragment.java

```diff
@@ -200,16 +204,24 @@ protected boolean openDetailView(Integer courseId) {
         return true;
     }
 
-    public void setCourseSchedule(List<CourseComponent> courseSchedule) {
+    public void setCourseSchedule(List<List<CourseComponent>> courseSchedule) {
         mCourseSchedule = courseSchedule;
         // The week view has infinite scrolling horizontally. We have to provide the events of a
         // month every time the month changes on the week view.
         weekView.setMonthChangeListener(new MonthLoader.MonthChangeListener() {
             @Override
             public List<? extends WeekViewEvent> onMonthChange(int newYear, int newMonth) {
                 List<WeekViewEvent> events = new ArrayList<>();
-                for (CourseComponent c : mCourseSchedule) {
-                    events.addAll(c.toWeekViewEvents(newMonth));
+                // Color each section and type a different color and add to view
+                Random rnd = new Random();
+                for (List<CourseComponent> components : mCourseSchedule) {
+                    int color = Color.argb(255, rnd.nextInt(256), rnd.nextInt(256), rnd.nextInt(256));
```

Comment:
yes, to peter's point - maybe instead of doing random, we can do a division. For example, if there are 5 components, take 255/5 and assign the first color to 51, second to 51*2, third to 51*3, ..., 255.